### ISSUE TITLE
Docstrings not accepted from parent class if parent is ABCMeta with Generic type

### ISSUE THREAD

**Issue by chbndrhnns** [PyCQA/pylint#4190]
### Steps to reproduce

<details>

<summary>Minimal working example</summary>

```py
"""Test"""
from abc import ABCMeta, abstractmethod
from typing import Generic, TypeVar

T = TypeVar("T")


class Base(Generic[T], metaclass=ABCMeta):
    """Base"""

    @classmethod
    @abstractmethod
    def do_something(cls):
        """Do something"""


class Impl(Base):
    """Impl"""

    def do_something(self):
        assert self


class ImplStr(Base[str]):
    """Impl"""

    @classmethod
    def do_something(cls):
        assert cls

```

</details>

### Current behavior

```
************* Module tests.test_p
tests/test_p.py:28:4: C0116: Missing function or method docstring (missing-function-docstring)

-------------------------------------------------------------------
Your code has been rated at 9.09/10 (previous run: 10.00/10, -0.91)
```

### Expected behavior

Annotation from parent class should be accepted even if the child is a parametrized version of the parent class.
As of now, only if the child is NOT parametrized, the parent docstring is accepted.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.7.2
astroid 2.5.1
Python 3.8.6 (default, Nov  4 2020, 11:22:47)
[Clang 10.0.1 (clang-1001.0.46.4)]
```
